ha_mroonga.cpp

```diff
@@ -14954,55 +14954,121 @@ mrn_io_and_cpu_cost ha_mroonga::scan_time()
+                                                 blocks);
     MRN_SET_BASE_SHARE_KEY(share, table->s);
     MRN_SET_BASE_TABLE_KEY(this, table);
   } else {
     MRN_SET_WRAP_SHARE_KEY(share, table->s);
     MRN_SET_WRAP_TABLE_KEY(this, table);
-    res = wrap_handler->read_time(index, ranges, rows);
+    res = wrap_handler->MRN_HANDLER_KEYREAD_TIME(index, ranges, rows, blocks);
     MRN_SET_BASE_SHARE_KEY(share, table->s);
     MRN_SET_BASE_TABLE_KEY(this, table);
   }
   DBUG_RETURN(res);
 }
 #endif
 
-double ha_mroonga::storage_read_time(uint index, uint ranges, ha_rows rows)
+mrn_io_and_cpu_cost ha_mroonga::storage_read_time(uint index,
+                                                  uint ranges,
+                                                  ha_rows rows,
+                                                  ulonglong blocks)
 {
   MRN_DBUG_ENTER_METHOD();
-  double time = handler::read_time(index, ranges, rows);
+  mrn_io_and_cpu_cost time =
+    handler::MRN_HANDLER_KEYREAD_TIME(index, ranges, rows, blocks);
   DBUG_RETURN(time);
 }
 
+#ifdef MRN_HANDLER_USE_KEYREAD_TIME
+IO_AND_CPU_COST ha_mroonga::keyread_time(uint index,
+                                         uint ranges,
+                                         ha_rows rows,
+                                         ulonglong blocks)
+#else
 double ha_mroonga::read_time(uint index, uint ranges, ha_rows rows)
+#endif
+{
+  MRN_DBUG_ENTER_METHOD();
+  mrn_io_and_cpu_cost time;
+#ifdef MRN_ENABLE_WRAPPER_MODE
+  if (share->wrapper_mode)
+  {
+    time = wrapper_read_time(index, ranges, rows
+#  ifdef MRN_HANDLER_USE_KEYREAD_TIME
+                             , blocks
+#  endif
+                             );
+  } else {
+#endif
+    time = storage_read_time(index, ranges, rows
+#ifdef MRN_HANDLER_USE_KEYREAD_TIME
+                             , blocks
+#endif
+                             );
+#ifdef MRN_ENABLE_WRAPPER_MODE
+  }
+#endif
+  DBUG_RETURN(time);
+}
+
+#ifdef MRN_HANDLER_USE_KEYREAD_TIME
+#  ifdef MRN_ENABLE_WRAPPER_MODE
+IO_AND_CPU_COST ha_mroonga::wrapper_rnd_pos_time(ha_rows rows)
+{
+  MRN_DBUG_ENTER_METHOD();
+  IO_AND_CPU_COST res;
+  MRN_SET_WRAP_SHARE_KEY(share, table->s);
+  MRN_SET_WRAP_TABLE_KEY(this, table);
+  res = wrap_handler->rnd_pos_time(rows);
+  MRN_SET_BASE_SHARE_KEY(share, table->s);
+  MRN_SET_BASE_TABLE_KEY(this, table);
+  DBUG_RETURN(res);
+}
+#  endif
+
+IO_AND_CPU_COST ha_mroonga::storage_rnd_pos_time(ha_rows rows)
 {
   MRN_DBUG_ENTER_METHOD();
-  double time;
+  IO_AND_CPU_COST res;
+  res = handler::rnd_pos_time(rows);
+  DBUG_RETURN(res);
+}
+
+IO_AND_CPU_COST ha_mroonga::rnd_pos_time(ha_rows rows)
+{
+  MRN_DBUG_ENTER_METHOD();
+  IO_AND_CPU_COST time;
 #ifdef MRN_ENABLE_WRAPPER_MODE
   if (share->wrapper_mode)
   {
-    time = wrapper_read_time(index, ranges, rows);
+    time = wrapper_rnd_pos_time(rows);
   } else {
 #endif
-    time = storage_read_time(index, ranges, rows);
+    //    time = storage_rnd_pos_time(rows);
```

Comment:
Is this comment necessary?

Reply:
Ah...
This is mistake. This comment is needless.

Reply:
I removed this comment.